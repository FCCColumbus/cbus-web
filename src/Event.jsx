function Event({ event, id }) {
    const href = `/coding-${id + 1}.jpg`

    return (
        <div className="event">
            <img src={href} alt='people coding' className="event-img" />
            <div className="info">
                <h5>{event.name}</h5>
                <table>
                    <tbody>
                        <tr>
                            <th>time:</th>
                            <td>{new Intl.DateTimeFormat('en-US', { dateStyle: 'medium', timeStyle: 'short' }).format(event.time)}</td>
                        </tr>
                        <tr>
                            <th>location:</th>
                            <td>{event.location}</td>
                        </tr>
                    </tbody>
                </table>
                <p>{event.description}</p>
            </div>
        </div>
    )
}

export default Event