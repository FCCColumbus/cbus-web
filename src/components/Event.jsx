import PropTypes from "prop-types";

function Event({ event }) {
  // add id to event
  // const href = `/coding-${id + 1}.jpg`
  const formattedDateTime = new Intl.DateTimeFormat('en-US', { dateStyle: 'medium', timeStyle: 'short' }).format(event.dtstart.value);
  console.log(event.location);

  /*
    We replace the "narrow non-breaking space" that appears before the AM/PM in some environments
    with a regular space to help prevent snapshot test failures.
    fwiw, it appears that this is a somewhat recent change in Node: https://www.reddit.com/r/webdev/comments/1054ve6/psa_node_1813_changed_datetime_formatting/
      - jedduffey@gmail.com, 2023 May 22
  */
  const replacedNonBreakingSpace = formattedDateTime.replace(/\u202f/g, ' ');
  
  return (
    <div className="event">
      {/* <img src={href} alt='people coding' className="event-img" /> */}
      <div className="event-info">
        <h5>{event.summary.value}</h5>
        <table>
          <tbody>
            <tr>
              <th>time:</th>
              <td>{replacedNonBreakingSpace}</td>
            </tr>
            <tr>
              <th>location:</th>
              {event.location && <td>{event.location.value}</td>}
            </tr>
          </tbody>
        </table>
        <p>{event.description.value}</p>
        <div className="meet-up-register">
          <a target="_blank" without rel="noreferrer" href="https://www.meetup.com/register/">Join Meetup.com</a>
        </div>
      </div>
    </div>
  )
}

Event.propTypes = {
  event: PropTypes.shape({
    summary: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    dtstart: PropTypes.objectOf(Date),
    location : PropTypes.string.isRequired,
    // id : PropTypes.number.isRequired,
  }).isRequired,
  // id: PropTypes.number.isRequired,
};

export default Event