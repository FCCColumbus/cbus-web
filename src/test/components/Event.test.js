import renderer from "react-test-renderer";

import Event from "../../Event";

describe("Event", () => {
    it("renders", () => {
        const event = {
            name: "Test Event",
            time: new Date('2023-04-17T18:30:00'),
            location: "Test Location",
            description: "This is a test event.",
        };

        const view = renderer.create(<Event event={event} />);
        expect(view).toMatchSnapshot();
    });
});
