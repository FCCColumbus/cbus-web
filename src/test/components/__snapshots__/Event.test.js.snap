// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders 1`] = `
<div
  className="event"
>
  <img
    alt="people coding"
    className="event-img"
    src="/coding-2.jpg"
  />
  <div
    className="info"
  >
    <h5>
      Test Event
    </h5>
    <table>
      <tbody>
        <tr>
          <th>
            time:
          </th>
          <td>
            Apr 17, 2023, 6:30 PM
          </td>
        </tr>
        <tr>
          <th>
            location:
          </th>
          <td>
            Test Location
          </td>
        </tr>
      </tbody>
    </table>
    <p>
      This is a test event.
    </p>
  </div>
</div>
`;
